src/test/java/com/teamsfinder/userwriteservice/user/service/UserServiceImpTest.java

```diff
@@ -0,0 +1,128 @@
+package com.teamsfinder.userwriteservice.user.service;
+
+import com.teamsfinder.userwriteservice.user.dto.EditUserDto;
+import com.teamsfinder.userwriteservice.user.dto.UserResponseDto;
+import com.teamsfinder.userwriteservice.user.exception.KeyCloakException;
+import com.teamsfinder.userwriteservice.user.exception.UserNotFoundException;
+import com.teamsfinder.userwriteservice.user.keycloak.KeyCloakService;
+import com.teamsfinder.userwriteservice.user.model.AccountType;
+import com.teamsfinder.userwriteservice.user.model.User;
+import com.teamsfinder.userwriteservice.user.repository.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceImpTest {
+    private static final String USER_KEYCLOAK_ID = "KEYCLOAKID";
+    private static final String EDIT_STRING = "EDITED";
+    private static final String USER_GITHUB = "GITHUB";
+    private static final String USER_PICTURE = "PICTURE";
+
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private KeyCloakService keyCloakService;
+
+    @InjectMocks
+    private UserServiceImp userService;
+
+    private User testUser = User.builder()
+            .id(1L)
+            .keyCloakId(USER_KEYCLOAK_ID)
+            .accountType(AccountType.USER)
+            .githubProfileUrl(USER_GITHUB)
+            .profilePictureUrl(USER_PICTURE)
+            .blocked(false)
+            .tags(new ArrayList<>())
+            .build();
+
+    @Test
+    void shouldCreateUser() {
+        //given
+        //when
+        Mockito.when(userRepository.save(Mockito.any(User.class))).thenReturn(testUser);
```

Comment:
brzydki mock zakłamujący rzeczywistość, polecam spojrzeć na doAnswer i z inwokacji zwrócić pierwszy argument, mocki idą do given

Reply:
Wytlumaczysz mi to na dc, bo nie za bardzo czaje co mam zmienic

Reply:
doAnswer(invocation -> invocation.getArgument(0)).when(userRepository.save(any(User.class)); coś w tym stylu, pisane z pamięci więc pewnie syntax lekko inny